lib/avromatic/model/types/decimal_type.rb

```diff
@@ -0,0 +1,58 @@
+# frozen_string_literal: true
+
+require 'bigdecimal'
+require 'bigdecimal/util'
+require 'avromatic/model/types/abstract_type'
+
+module Avromatic
+  module Model
+    module Types
+      class DecimalType < AbstractType
+        VALUE_CLASSES = [::Numeric].freeze
+
+        attr_reader :precision, :scale
+
+        def initialize(precision:, scale: 0)
+          super()
+          @precision = precision
+          @scale = scale
+        end
+
+        def value_classes
+          VALUE_CLASSES
+        end
+
+        def name
+          "decimal(#{precision}, #{scale})"
+        end
+
+        def coerce(input)
+          case input
+          when ::NilClass, ::BigDecimal
+            input
+          when ::Numeric
```

Comment:
Can we restrict this conditional to just `Float` and `Integer` since we're not supporting `Rational` or `Complex`? The implementation of `coercible?` will also require a similar change.

Reply:
@jturkel Should the `VALUE_CLASSES` and `INPUT_CLASSES` be limited to `[BigDecimal, Float, Integer]` as well then?

Reply:
`DecimalType#input_classes` should return `[BigDecimal, Float, Integer]` because those are the types that we support as input for decimal attributes in constructors and attribute setters. `DecimalType#value_classes` should return `[BigDecimal]` because decimal attribute values are always coerced to a `BigDecimal` and a decimal attribute getter will always return a `BigDecimal`.

Reply:
Make sense. Updated.